packages/react-core/src/components/TreeView/examples/TreeViewCheckbox.tsx

```diff
@@ -0,0 +1,191 @@
+              id: 'example3-App2Settings',
+              checkProps: { checked: false }
+            },
+            {
+              name: 'Loader',
+              id: 'example3-App2Loader',
+              checkProps: { checked: false },
+              children: [
+                {
+                  name: 'Loading App 1',
+                  id: 'example3-LoadApp1',
+                  checkProps: { checked: false }
+                },
+                {
+                  name: 'Loading App 2',
+                  id: 'example3-LoadApp2',
+                  checkProps: { checked: false }
+                },
+                {
+                  name: 'Loading App 3',
+                  id: 'example3-LoadApp3',
+                  checkProps: { checked: false }
+                }
+              ]
+            }
+          ]
+        }
+      ],
+      defaultExpanded: true
+    },
+    {
+      name: 'Cost management',
+      id: 'example3-Cost',
+      checkProps: { 'aria-label': 'cost-check', checked: false },
+      children: [
+        {
+          name: 'Application 3',
+          id: 'example3-App3',
+          checkProps: { 'aria-label': 'app-3-check', checked: false },
+          children: [
+            {
+              name: 'Settings',
+              id: 'example3-App3Settings',
+              checkProps: { 'aria-label': 'app-3-settings-check', checked: false }
+            },
+            {
+              name: 'Current',
+              id: 'example3-App3Current',
+              checkProps: { 'aria-label': 'app-3-current-check', checked: false }
+            }
+          ]
+        }
+      ]
+    },
+    {
+      name: 'Sources',
+      id: 'example3-Sources',
+      checkProps: { 'aria-label': 'sources-check', checked: false },
+      children: [
+        {
+          name: 'Application 4',
+          id: 'example3-App4',
+          checkProps: { 'aria-label': 'app-4-check', checked: false },
+          children: [
+            {
+              name: 'Settings',
+              id: 'example3-App4Settings',
+              checkProps: { 'aria-label': 'app-4-settings-check', checked: false }
+            }
+          ]
+        }
+      ]
+    },
+    {
+      name: 'Really really really long folder name that overflows the container it is in',
+      id: 'example3-Long',
+      checkProps: { 'aria-label': 'long-check', checked: false },
+      children: [
+        { name: 'Application 5', id: 'example3-App5', checkProps: { 'aria-label': 'app-5-check', checked: false } }
+      ]
+    }
+  ];
+
+  const [checkedItems, setCheckedItems] = React.useState<TreeViewDataItem[]>([]);
+
+  const onCheck = (event: React.ChangeEvent, treeViewItem: TreeViewDataItem) => {
+    const checked = (event.target as HTMLInputElement).checked;
+
+    const checkedItemTree = options.map(opt => Object.assign({}, opt)).filter(item => filterItems(item, treeViewItem));
+    const flatCheckedItems = flattenTree(checkedItemTree);
+
+    setCheckedItems(
```

Comment:
I think you should use an updater function for the current state of `checkedItems` here as well.

Reply:
I'm not entirely sure how to do that since it relies on the "checked" boolean

Reply:
🤔 are you not able to access that variable from within an updater function?